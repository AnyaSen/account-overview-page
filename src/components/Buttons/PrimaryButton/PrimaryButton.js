import React from "react";

export default function PrimaryButton({ btnText }) {
  return <button>{btnText}</button>;
}
