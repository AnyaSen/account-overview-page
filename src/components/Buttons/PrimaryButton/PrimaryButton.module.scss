@import '../../../assets/scss/mixins';

.PrimaryButton {
	@include button($color-green, $color-dark-green, $color-white);
}
