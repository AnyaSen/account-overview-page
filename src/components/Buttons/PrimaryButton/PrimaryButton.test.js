import React from "react";
import { render, cleanup } from "@testing-library/react";

import PrimaryButton from "./PrimaryButton";

describe("<PrimaryButton />", () => {
  afterAll(cleanup);

  test("Should have correct text and be disabled.", () => {
    const text = "Search";
    const isDisabled = true;

    const { getByText, container } = render(
      <PrimaryButton btnText={text} isDisabled={isDisabled} />
    );

    const button = getByText(text);

    expect(button).toBeTruthy();
    expect(button).toHaveAttribute("disabled");

    expect(container.firstChild).toMatchSnapshot();
  });
});
