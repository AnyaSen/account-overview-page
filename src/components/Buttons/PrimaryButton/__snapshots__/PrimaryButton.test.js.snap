// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimaryButton /> Should have correct text and be disabled. 1`] = `
<button
  class="PrimaryButton"
  disabled=""
  type="submit"
  value="submit"
>
  Search
</button>
`;
